transaction-view/src/address_table_lookup_meta.rs

```diff
@@ -103,6 +104,76 @@ impl AddressTableLookupMeta {
     }
 }
 
+pub struct AddressTableLookupIterator<'a> {
+    pub(crate) bytes: &'a [u8],
+    pub(crate) offset: usize,
+    pub(crate) num_address_table_lookup: u8,
+    pub(crate) index: u8,
+}
+
+impl<'a> Iterator for AddressTableLookupIterator<'a> {
+    type Item = SVMMessageAddressTableLookup<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.index < self.num_address_table_lookup {
+            // Each ATL has 3 pieces:
+            // 1. Address (Pubkey)
+            // 2. write indexes ([u8])
+            // 3. read indexes ([u8])
+
+            // Advance offset for address of the lookup table.
+            const _: () = assert!(core::mem::align_of::<Pubkey>() == 1, "Pubkey alignment");
+            // SAFETY:
+            // - The offset is checked to be valid in the slice.
+            // - The alignment of Pubkey is 1.
+            // - `Pubkey` is a byte array, it cannot be improperly initialized.
+            let account_key = unsafe { read_type::<Pubkey>(self.bytes, &mut self.offset) }.ok()?;
```

Comment:
nit: the iterator is created after the layout of the ATLs has been already checked
from `AddressTableLookupMeta` right? So you could probably squeeze some extra
perf out of this by removing the check_remaining() calls and instead asserting
things at the start of next(). Since most of these functions are inline, chances
are that asserting things will elide bound checks in the slice::get calls made
by optimized_read_compressed_u16.

Reply:
Kept things with checks - if want to do an "unsafe" optimization like that I need a bench to convince me its' worth it 😄 
I agree with the ideas though!